docs/topics/compatibility-guide-19.md

```diff
@@ -41,102 +41,92 @@ perspective
 > - 1.8.0: raise the warning to an error
 -->
 
-### TODO: KT-45375
-
-> **Issue**: [KT-45375](https://youtrack.jetbrains.com/issue/KT-45375)
->
-> **Component**: Kotlin/JVM
->
-> **Incompatible change type**: source
->
-> **Short summary**:
->
-> **Deprecation cycle**:
-
-### TODO: KT-46344
+### Prohibit super class constructor call on a function interface in supertypes list
 
 > **Issue**: [KT-46344](https://youtrack.jetbrains.com/issue/KT-46344)
 >
 > **Component**: Core language
 >
 > **Incompatible change type**: source
 >
-> **Short summary**:
+> **Short summary**: Kotlin 1.9 prohibits super class constructor calls on functional supertypes.
 >
 > **Deprecation cycle**:
+> * 1.7.0: report a warning (or an error in progressive mode) on constructor calls on functional supertypes
+> * 1.9.0: raise the warning to an error
 
-### TODO: KT-47932
+### Prohibit cycles in annotations
 
 > **Issue**: [KT-47932](https://youtrack.jetbrains.com/issue/KT-47932)
 >
 > **Component**: Core language
 >
 > **Incompatible change type**: source
 >
-> **Short summary**:
+> **Short summary**: Kotlin 1.9 prohibits cycles in types of annotation parameters, unless the type is an `Array` or `vararg`.
```

Comment:
I've updated to:

> Kotlin 1.9 prohibits the type of an annotation being used as one of its parameter types, either directly or indirectly. This prevents cycles from being created.
> The only exception is if the parameter type is an `Array` or `vararg` of the annotation type.

Is this better?

Reply:
Yes, but I'd rephrase the last sentence without "exception" as it may have another meaning of a runtime exception.
Like "However, using the parameter type that is an `Array` or `vararg` of the annotation type is still allowed"